src/rpdk/core/contract/resource_client.py

```diff
@@ -287,9 +287,25 @@ def assert_primary_identifier_not_updated(
             == traverse(updated_model, fragment_list(primary_identifier, "properties"))[
                 0
             ]
-            for primary_identifier in list(primary_identifier_path)
+            for primary_identifier in primary_identifier_path
         )
 
+    @staticmethod
+    def does_primary_identifier_exist(
```

Comment:
isnt this the same as assert_primary_identifier other than the assert? the former ensures all primary identifiers are present and asserts on that. this one just returns true false. could maybe combine and use the `assert_primary_identifier` logic here and have `assert_primary_identifier` just assert on `does_primary_identifier_exist`

Reply:
this method name doesnt seem to match what is occurring,. this simply checks if the updated and created model have the same values for primary identifier, not if they exist or not. if the two models both have a value of none for the pid, it would pass

Reply:
renaming this method to ```assert_primary_identifier_equal``` and removing the method ```assert_primary_identifier_not_updated``` to avoid replication